File: Composicion/ejercicio 1/ej1.py
class Habitacion:
    def __init__(self, nombre, tamano): 
        self._nombre = nombre
        self._tamano = tamano

    @property
    def nombre(self):
        return self._nombre

    @nombre.setter
    def nombre(self, valor):
        self._nombre = valor

    @property
    def tamano(self):
        return self._tamano

    @tamano.setter
    def tamano(self, valor):
        self._tamano = valor

    def mostrar_info(self):
        print(f"Habitación: {self.nombre}, Tamaño: {self.tamano} m²")

class Casa:
    def __init__(self, direccion):  
        self._direccion = direccion
        self._habitaciones = []

    @property
    def direccion(self):
        return self._direccion

    @direccion.setter
    def direccion(self, valor):
        self._direccion = valor

    def agregar_habitacion(self, habitacion):
        if isinstance(habitacion, Habitacion):
            self._habitaciones.append(habitacion)

    def mostrar_casa(self):
        print(f"\n--- Casa en {self.direccion} ---")
        if not self._habitaciones:
            print("No hay habitaciones registradas.")
        for hab in self._habitaciones:
            hab.mostrar_info()

if __name__ == "__main__":
    casa = Casa("Calle Ruiz Orellana 1233")
    casa.agregar_habitacion(Habitacion("Sala", 25))
    casa.agregar_habitacion(Habitacion("Cocina", 12))
    casa.agregar_habitacion(Habitacion("Dormitorio", 18))
    casa.mostrar_casa()

File: Composicion/ejercicio 1/ej1.java
import java.util.ArrayList;

// Clase Habitacion
class Habitacion {
    private String nombre;
    private double tamano;

    public Habitacion(String nombre, double tamano) {
        this.nombre = nombre;
        this.tamano = tamano;
    }

    public String getNombre() {
        return nombre;
    }

    public void setNombre(String nombre) {
        this.nombre = nombre;
    }

    public double getTamano() {
        return tamano;
    }

    public void setTamano(double tamano) {
        this.tamano = tamano;
    }

    public void mostrarInfo() {
        System.out.println("Habitación: " + nombre + ", Tamaño: " + tamano + " m²");
    }
}

// Clase Casa
class Casa {
    private String direccion;
    private ArrayList<Habitacion> habitaciones;

    public Casa(String direccion) {
        this.direccion = direccion;
        this.habitaciones = new ArrayList<>();
    }

    public String getDireccion() {
        return direccion;
    }

    public void setDireccion(String direccion) {
        this.direccion = direccion;
    }

    public void agregarHabitacion(Habitacion habitacion) {
        habitaciones.add(habitacion);
    }

    public void mostrarCasa() {
        System.out.println("\n--- Casa en " + direccion + " ---");
        if (habitaciones.isEmpty()) {
            System.out.println("No hay habitaciones registradas.");
        } else {
            for (Habitacion h : habitaciones) {
                h.mostrarInfo();
            }
        }
    }
}

public class Main {
    public static void main(String[] args) {
        Casa casa = new Casa("Calle Ruiz Orellana 1233");
        casa.agregarHabitacion(new Habitacion("Sala", 25));
        casa.agregarHabitacion(new Habitacion("Cocina", 12));
        casa.agregarHabitacion(new Habitacion("Dormitorio", 18));
        casa.mostrarCasa();
    }
}